pkg/config/generate_test.go

```diff
@@ -26,39 +27,162 @@ properties:
         PublicSubdomain: ""
 `)
 
+func TestGenerate(t *testing.T) {
+	tests := map[string]struct {
+		omc *api.OpenShiftManagedCluster
+	}{
+		"test generate new": {
+			omc: testOpenShiftCluster(),
+		},
+		//TODO "test generate doesn't overwrite": {},
+	}
+
+	for name, test := range tests {
+		err := Generate(test.omc)
+		if err != nil {
+			t.Errorf("%s received generation error %v", name, err)
+			continue
+		}
+		testRequiredFields(test.omc, t)
+		// check mutation
+	}
+}
+
+func testRequiredFields(omc *api.OpenShiftManagedCluster, t *testing.T) {
+	checkStringField := func(f string, name string) {
```

Comment:
the top 3 of these condense down to `assert := func(b bool, msg string) { if(!b) { t.Error(msg) } }`